### PR TITLE
Refs #25163, #25165, #25166 - expose macros and properties in safemode

### PR DESCRIPTION
**Pull request by mbacovsky**
This PR add macros and attributes necessary for new report templates.

### CODE REVIEW

File: app/lib/katello/concerns/base_template_scope_extensions.rb [hunk 1]
```diff
@@ -5,7 +5,8 @@ module BaseTemplateScopeExtensions
 
       module Overrides
         def allowed_helpers
-          super + [:errata]
+          super + [:errata, :host_subscriptions, :host_applicable_errata_ids, :host_applicable_errata_filtered,
+            :host_latest_applicable_rpm_version, :load_pools]
```
Comment (ares) [inline app/lib/katello/concerns/base_template_scope_extensions.rb]: rubocop is unhappy on this line
```
/var/lib/workspace/workspace/katello-pr-test/app/lib/katello/concerns/base_template_scope_extensions.rb:9:13: C: 
Layout/AlignArray: Align the elements of an array literal if they span more than one line.

            :host_latest_applicable_rpm_version, :load_pools]

            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by jlsherrill** [comment]
We have triaged this in our weekly katello triage and decided to target this for 3.10.  Because of this, we cannot merge this until after 3.9 has branched which should be sometime in the next week.

**Comment by ares** [comment]
@jsherill would you mind reconsiderring? This should get to 3.9 in order we can include Katello reports into Foreman 1.20. I think this is very low risk, I've tested this already and it works fine.

**Comment by ares** [comment]
Also I wonder, given this does not remove or change any existing functionality, why this can't be merged during stabilization week?

**Comment by jlsherrill** [comment]
@ares sure thing, retriaged for 3.9

**Comment by mbacovsky** [comment]
@ares, updated.

**Comment by ares** [comment]
I think you'll need to shorten the git message. Perhaps "Refs #25163, #25165, #25166 - report macros" would be good enough :-) Otherwise :+1: 

If someone could quickly check the sanity of queries, that would be good. Given they are now abstracted, we can always change them later without causing backwards incompatibility in our templates.

File: app/lib/katello/concerns/base_template_scope_extensions.rb [hunk 2]
```diff
@@ -16,6 +17,26 @@ def allowed_helpers
...
+      def host_applicable_errata_ids(host)
+        host.applicable_errata.map(&:errata_id)
+      end
+
+      def host_applicable_errata_filtered(host, filter = '')
+        host.applicable_errata.search_for(filter)
+      end
+
+      def host_latest_applicable_rpm_version(host, package)
+        host.applicable_rpms.where('name = ?', package).sort_by(&:version_sortable).first.try(:nvra)
```
Comment (stbenjam) [inline app/lib/katello/concerns/base_template_scope_extensions.rb]: This is somewhat inefficient.  Why not `host.applicable_rpms.where('name = ?',  package).order(:version_sortable).limit(1).pluck(:nvra).first`?

**Comment by stbenjam** [comment]
But other than one comment above this looks good

File: app/lib/katello/concerns/base_template_scope_extensions.rb (hunk 2 above)
Comment (ares) [reply app/lib/katello/concerns/base_template_scope_extensions.rb]: :+1: while at it, `where(name: package)` would also be nicer

**Pull request merged**